### ISSUE TITLE
Add support for ES 2.0 pipeline aggregations

### ISSUE THREAD

**Issue by extesy** [elastic/elasticsearch-dsl-py#247]
Elasticsearch 2.0-beta now supports parent and sibling aggregations: https://www.elastic.co/blog/out-of-this-world-aggregations and http://amsterdam.luminis.eu/2015/08/31/elasticsearch-2-0-and-pipeline-aggregations/

It looks like I can't use them using elasticsearch-dsl at the moment since `.bucket()` call always creates a nested aggregation while I need a sibling one.

**Comment by ei-grad** [comment]
Just my 50 cents - for now one could just add a sibling aggregation explicitly:
```python
s = Search()
s.aggs.bucket(
    "target", "date_histogram",
    interval="hour", field="@timestamp"
)
s.aggs['target'].bucket(
    "movavg", "moving_avg",
    buckets_path="hau",
    model="holt_winters", window=90 * 24,
    minimize=True, settings={'period': 7 * 24},
    predict=14 * 24,
)
s.aggs['target'].metric(
    "hau", "cardinality", field="distinct_id"
)
```

I doubt how it could be look better in implicit way. May be just add all new aggregations as a bucket aggragations?

**Comment by HonzaKral** [comment]
Fixed by cba1d9e
